File: app/src/main/java/ssilvalucas/testeandroidv2/screen/home/HomeInteractor.java
package ssilvalucas.testeandroidv2.screen.home;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import ssilvalucas.testeandroidv2.data.api.StatementsService;
import ssilvalucas.testeandroidv2.data.model.StatementsResponse;
import ssilvalucas.testeandroidv2.data.retrofit.Retrofit;

interface HomeInteractorInput{
    void fetchStatementsById(long id);
}

public class HomeInteractor implements HomeInteractorInput{

    public HomePresenterInput output;

    @Override
    public void fetchStatementsById(long id) {
        StatementsService statementsService = Retrofit.getInstance().create(StatementsService.class);

        statementsService.getStatements(id).enqueue(new Callback<StatementsResponse>() {
            @Override
            public void onResponse(Call<StatementsResponse> call, Response<StatementsResponse> response) {
                if(response.isSuccessful()){
                    output.onSuccessfulFetchStatements(response.body());
                }
            }

            @Override
            public void onFailure(Call<StatementsResponse> call, Throwable t) {
                output.throwError(t.getMessage());
            }
        });
    }
}

File: dev/SantanTest/app/src/main/java/com/example/santantest/domain/interactor/home/HomeInteractor.kt
package com.example.santantest.domain.interactor.home

import android.util.Log
import com.example.santantest.data.AccountService
import com.example.santantest.data.Network
import com.example.santantest.domain.model.StatementResponse
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class HomeInteractor {

    fun getStatements(userID: Int, callback: HomeInteractorListener){
        val accountService = Network.getInstance()?.create(AccountService::class.java)
        val res = accountService?.getStatements(userID)
        res?.enqueue(object : Callback<StatementResponse>{

            override fun onFailure(call: Call<StatementResponse>, t: Throwable) {
                Log.e("ERRO",t.message)
            }

            override fun onResponse(
                call: Call<StatementResponse>,
                response: Response<StatementResponse>
            ) {
                response.body()?.statementList?.let {
                    callback.onGetStatementsSuccess(it)
                }
            }
        })



    }

}